### DISCUSSION TITLE
Controls for standalone deckgl application

### DISCUSSION THREAD

**Discussion by geoyogesh** [visgl/deck.gl#5354] [Ideas]
I am migrating existing mapping application to standalone deckgl app without other mapping applications like mapbox, google maps etc

I thought it would be really useful to provide essential controls as part of deckgl. below are some controls i have in mind. 

1. Navigation contol - Zoom In, Zoom Out and (altitude, bearing, pitch) control
2. Home Control
3. FullScreen
4. Scale Bar
5. Attribution Control
6. Geolocate Control
7. Basemap switcher Control
8. Busy or Loading Control -> to display progress during network calls 
9. Sketch Control -> edit Point, LineString, Polygon, MultiPoint, MultiLineString, and MultiPolygon (geometries supported by geojson spec)
10. Annotate 
11. Print
12. Measure
13. Legend
14. Overview Control

list can be extended to to include more features such as layer toggle, opacity control and styling etc

**Comment by ibgreen** [comment]
This is a nice idea. Some initial quick thoughts:

- Ideally these components should be created as pure JavaScript components
- And have a declarative interface `new NavigationControl(props: {deck: Deck, ...})`. 
- This so that other bindings like `@deck.gl/json` and `pydeck` can use them
- These components would presumably go into a separate package. `@deck.gl/html-controls`?
- It would of course also be very nice to have react wrappers of these components.

If someone wanted to do a proof of concept without hashing out too many details in advance, these controls could probably initially be created in an independent repository and then merged into deck.gl.
